astropy/wcs/src/unit_list_proxy.c

```diff
@@ -186,6 +188,44 @@ PyUnitListProxy_getitem(
   return result;
 }
 
+static PyObject *
+PyUnitListProxy_richcmp(PyUnitListProxy *self, PyUnitListProxy *other, int op)
+{int equal;
+  int status;
+   
+
+  if ((op == Py_EQ || op == Py_NE) &&
+      PyObject_TypeCheck(other, &PyUnitListProxyType)) {
+     
+
+    if (self->size!=other->size||self->unit_class!=other->unit_class||self->array!=other->array)
+       {status = 0;
+        equal = 0;}
+    else if (self == 0x0 || other == 0x0){
+       status = 1;}
+    else{status = 0;
+         equal = 1;}
+
+    if (status == 0) {
+      if (op == Py_NE) {
+        equal = !equal;
+      }
+      if (equal) {
+        Py_RETURN_TRUE;
+      } else {
+        Py_RETURN_FALSE;
+      }
+    } else {
+      return NULL;
```

Comment:
Returning `NULL` without an exception will probably result in a `SystemError` - or is that handled differently for `richcmp`? Did you check this path?

By the way I would prefer it to immediately fail in case an exception should be raised. That way one could also remove the `status` variable completely.